py/torch_tensorrt/dynamo/utils.py

```diff
@@ -661,3 +661,57 @@ def get_flat_args_with_check(
     flat_args_with_path, received_spec = pytree.tree_flatten_with_path((args, kwargs))
     flat_args = tuple(x[1] for x in flat_args_with_path)
     return flat_args, received_spec
+
+
+def get_output_meta_val(output: Any) -> List[Any]:
+    output_meta_val = []
+    if isinstance(output, torch.fx.node.Node):
+        if "val" in output.meta:
+            output_meta_val.append(output.meta["val"])
+    elif isinstance(output, (tuple, list)):
+        for node in output:
+            output_meta_val.extend(get_output_meta_val(node))
+    else:
+        raise ValueError(
+            f"expect torch.fx.node.Node or a tuple/list of torch.fx.node.Node type, got unexpected types: {type(output)=}"
+        )
+    return output_meta_val
+
+
+def set_output_meta_val(output: Any, outputs_meta_val: List[Any]) -> None:
+    if isinstance(output, torch.fx.node.Node):
+        assert len(outputs_meta_val) > 0
+        if "val" not in output.meta:
+            output.meta["val"] = outputs_meta_val[0]
+        outputs_meta_val.pop(0)
+    elif isinstance(output, (tuple, list)):
+        for node in output:
+            set_output_meta_val(node, outputs_meta_val)
+    else:
+        raise ValueError(
+            f"expect torch.fx.node.Node or a tuple/list of torch.fx.node.Node type, got unexpected types: {type(output)=}"
+        )
+
+
+def get_output_dtypes(output: Any, truncate_doulbe: bool = False) -> List[dtype]:
+    output_dtypes = []
+    if isinstance(output, torch.fx.node.Node):
+        if "val" in output.meta:
+            output_meta = output.meta["val"]
+            if isinstance(output_meta, (FakeTensor, torch.Tensor)):
+                if truncate_doulbe and output_meta.dtype == torch.float64:
+                    output_dtypes.append(dtype.float32)
+                else:
+                    output_dtypes.append(dtype._from(output_meta.dtype))
+        else:
```

Comment:
also check "tensor_meta" field in output.meta

Reply:
added as suggested